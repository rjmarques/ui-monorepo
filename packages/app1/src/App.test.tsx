import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';

import App from 'App';

describe('Renders main page correctly', async () => {
    it('Should render the page correctly', async () => {
        // Setup
        const result = render(<App />);
        const h1 = await screen.queryByText('App1: Vite + React');

        // Expectations
        expect(h1).not.toBeNull();

        // Snapshot
        expect(result).toMatchSnapshot()
    });
});