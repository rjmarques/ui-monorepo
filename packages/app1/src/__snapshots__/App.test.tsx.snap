// Vitest Snapshot v1

exports[`Renders main page correctly > Should render the page correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div>
          <a
            href="https://vitejs.dev"
            target="_blank"
          >
            <img
              alt="Vite logo"
              class="logo"
              src="/vite.svg"
            />
          </a>
          <a
            href="https://reactjs.org"
            target="_blank"
          >
            <img
              alt="React logo"
              class="logo react"
              src="/src/assets/react.svg"
            />
          </a>
        </div>
        <h1>
          App1: Vite + React
        </h1>
        <div
          class="card"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-diro6f ant-btn-primary ant-btn-dangerous"
            type="button"
          >
            <span>
              count is 0
            </span>
          </button>
          <p>
            Edit 
            <code>
              src/App.tsx
            </code>
             and save to test HMR, boy
          </p>
        </div>
        <p
          class="read-the-docs"
        >
          Click on the Vite and React logos to learn more
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div>
        <a
          href="https://vitejs.dev"
          target="_blank"
        >
          <img
            alt="Vite logo"
            class="logo"
            src="/vite.svg"
          />
        </a>
        <a
          href="https://reactjs.org"
          target="_blank"
        >
          <img
            alt="React logo"
            class="logo react"
            src="/src/assets/react.svg"
          />
        </a>
      </div>
      <h1>
        App1: Vite + React
      </h1>
      <div
        class="card"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-diro6f ant-btn-primary ant-btn-dangerous"
          type="button"
        >
          <span>
            count is 0
          </span>
        </button>
        <p>
          Edit 
          <code>
            src/App.tsx
          </code>
           and save to test HMR, boy
        </p>
      </div>
      <p
        class="read-the-docs"
      >
        Click on the Vite and React logos to learn more
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
